packages/react/src/Hero/Hero.tsx

```diff
@@ -74,7 +74,9 @@ function HeroDescription({size = '400', weight, children}: PropsWithChildren<Her
 
 type HeroActions = {
   href: string
-} & Omit<ButtonBaseProps, 'variant'>
+  leadingVisual?: React.ReactNode
```

Comment:
Can these be picked from ButtonBaseProps instead? And any ideas why this isn't getting inherited? Feels redundant to maintain these props here too.

Reply:
I left a comment on this https://github.com/primer/brand/pull/284#discussion_r1200626396 - I'm not entirely sure why it's throwing an error here. I get the following TS error: `Type 'FunctionComponent<IconProps>' is missing the following properties from type 'ReactElement<any, string | JSXElementConstructor<any>>': type, props, key`, which isn't present when utilizing `Button` and the "visual" props directly.

Utilizing `React.ReactNode` fixed it, as it seems to be tied to `ReactElement`. I kept the change in this component via `Omit` as to not touch `Button`. I can do some more digging in the meantime though.

Reply:
@rezrah - Digging into this a bit deeper it seems like this is tied to the `ReactElement` type in `Button` when utilizing an Octicon. ReactElement does not appear to apply when the type is a functional component `React.FC`, which is what `Octicon` seems to return as its type.

When used directly, `Button` doesn't seem to check the types of the props that are passed to it, which I think is why it doesn't cause an issue when `Button` is used directly with `leadingVisual` and `trailingVisual`. 

Could we replace the `ReactElement` type in `Button.tsx` with `React.ReactNode` or add both, (e.g. 
 `leadingVisual?: ReactElement | React.ReactNode`)? It seems like we do this within `Label` and `TextInput` too.
 
 Let me know your thoughts!

Reply:
Hey Tyler, you're right that it needs to be fixed in Button. Putting this here for posterity, as we spoke about this over slack. The type needs to extend Icon from Octicons, doing that should fix it. I've pushed the change in [this commit](https://github.com/primer/brand/pull/284/commits/2a9a062f645cf71c7d687d9d6aa9f7dfd992a9ba).